src/controls/search.js

```diff
@@ -233,13 +234,23 @@ const Search = function Search(options = {}) {
     const group = {};
     const ids = Object.keys(data);
     ids.forEach((id) => {
-      const item = data[id];
-      const type = item[layerNameAttribute];
-      if (type in group === false) {
-        group[type] = [];
-        item.header = viewer.getLayer(type).get('title');
+      const item = data[id];      
+      const typeTitle =
+        (layerNameAttribute && idAttribute) ? viewer.getLayer(item[layerNameAttribute]).get('title') :
```

Comment:
Could this if statement be refactored?

Reply:
Thanks for the review! I have pushed a commit refactoring the ternary assignment. Please take another look :)